crates/breez-sdk/lnurl/src/auth.rs

```diff
@@ -0,0 +1,71 @@
+use axum::{
+    extract::{self, Request},
+    http::{StatusCode, header::AUTHORIZATION},
+    middleware::Next,
+    response::Response,
+};
+use base64::{Engine as _, engine::general_purpose::STANDARD as BASE64};
+use tracing::{debug, error};
+use x509_parser::prelude::{FromDer, X509Certificate};
+
+use crate::state::State;
+
+pub async fn auth<DB>(
+    extract::State(state): extract::State<State<DB>>,
+    req: Request,
+    next: Next,
+) -> Result<Response, StatusCode>
+where
+    DB: Send + Sync + 'static,
+{
+    let Some(ca_cert) = state.ca_cert.as_ref() else {
+        return Ok(next.run(req).await);
+    };
+
+    let Ok((_, ca_cert)) = X509Certificate::from_der(ca_cert.as_ref()) else {
```

Comment:
Is is possible to store the X509Certificate in the state? Then we don't need to convert it for each request